test/detect/index.js

```diff
@@ -15,18 +14,19 @@ function testAutoDetection(language) {
   const languagePath = utility.buildPath('detect', language);
 
   it(`should have test for ${language}`, function() {
-    return fs.statAsync(languagePath)
+    return fs.stat(languagePath)
       .then(path => path.isDirectory().should.be.true);
   });
 
-  it(`should be detected as ${language}`, function() {
-    return fs.readdirAsync(languagePath)
+  it(`should be detected as ${language}`, async function() {
+    let dirs = await fs.readdir(languagePath)
```

Comment:
Only use `let` for reassignment. 

```suggestion
    const dirs = await fs.readdir(languagePath)
```